komorebi/src/window_manager.rs

```diff
@@ -747,34 +747,90 @@ impl WindowManager {
         Ok(())
     }
 
+    /// Calculates the direction of a move across monitors given a specific monitor index.
+    /// If multiple monitors are found, tries to find the one that makes the most sense to move the focused window to.
     pub fn monitor_idx_in_direction(&self, direction: OperationDirection) -> Option<usize> {
-        let current_monitor_size = self.focused_monitor_size().ok()?;
+        let current_monitor_size = match self.focused_monitor_size() {
+            Ok(size) => size,
+            Err(_) => return None,
+        };
+
+        let mut found: Vec<(usize, &Monitor)> = Vec::new();
```

Comment:
Let's declare something like `let mut target_idx = None` here, and then below replace all the returns with `target_idx = Some(whatever_index)`, and then finally replace the `None` returned at the end with `target_idx`